notes/programming-languages/python/packages/pandas.md

```diff
@@ -0,0 +1,123 @@
+# Python Language Overview
+
+### The `pandas` Module
+
+The `pandas` module is an extremely useful one for working with structured data. You can think of `pandas` as a python tool for creating and manipulating powerful spreadsheets-like objects called DataFrames. `pandas` also includes some SQL-like features and can be used to easily read and write data stored in csv's or databases.
+
+Reference:
+
+  + http://pandas.pydata.org/pandas-docs/stable/
+
+> NOTE: `pandas` supports Python 2.7 & 3.4+
+
+First install the package using pip, if necessary:
+
+```` sh
+# install requests
+pip install pandas
+````
+
+#### Reading data into DataFrames using `pandas`
+
+From the command-line:
+
+```shell
+python
+Python 3.5.2 |Anaconda 4.2.0 (64-bit)| (default, Jul  5 2016, 11:41:13) [MSC v.1900 64 bit (AMD64)] on win32
+Type "help", "copyright", "credits" or "license" for more information.
+>>> import pandas as pd # pd now references the pandas package, saving you some typing
+# We can create pandas DataFrames with native python data structures like lists and dictionaries
+>>> test = [[1,'a'], [2,'b'], [3,'c']]
+>>> df = pd.DataFrame(test)
+>>> df
+   0  1
+0  1  a
+1  2  b
+2  3  c
+# The columns have default numerical names. To change this, we can use the following:
+>>> df.columns = ['num', 'letter']
+>>> df
+   num letter
+0    1      a
+1    2      b
+2    3      c
+#Pandas also offers useful functions that read Excel and csv files directly into Dataframes, complete with column headings
+>>> stats = pd.read_excel(r'C:\Users\Mike\Desktop\jeter_stats.xlsx') # read a pre-saved file called 'jeter_stats'- you can use your own
```

Comment:
is the `jeter_stats.xlsx` excel file small enough / simple enough to include inside this repo? It would be nice to give students an example file to use, and I think they would find this instructive and appreciate the ability to use it for demonstration purposes.

if so, we could either add it somewhere near this package overview (maybe a bit out of place), or we can create a new ungraded exercise at `/exercises/pandas-practice/` (preferred) to contain the data file (`/exercises/pandas-practice/jeter_stats.xlsx`) and some of these file-specific instructions (`/exercises/pandas-practice/exercise.md`). then we can reference this package overview as a prerequisite for doing the exercise.

Is the exercise approach agreeable?